server/server.go

```diff
@@ -469,43 +469,81 @@ func New(
 	return s, nil
 }
 
-func (s *Server) getHostIP(configIP string) net.IP {
+func (s *Server) getHostIPs(configIP string) []string {
 	// emulate kubelet behavior of choosing hostIP
 	// ref: k8s/pkg/kubelet/nodestatus/setters.go
+	rootIP := ""
 
 	// use configured value if set
 	if hostIP := net.ParseIP(configIP); hostIP != nil {
-		return hostIP
+		rootIP = hostIP.String()
 	}
 
 	// Otherwise use kubernetes utility to choose hostIP
 	// there exists the chance for both hostIP and err to be nil, so check both
-	if hostIP, err := knet.ChooseHostInterface(); err != nil && hostIP != nil {
-		return hostIP
+	if rootIP == "" {
+		if hostIP, err := knet.ChooseHostInterface(); err != nil && hostIP != nil {
+			rootIP = hostIP.String()
+		}
 	}
 
 	// attempt to find an IP from the hostname
-	if hostname, err := os.Hostname(); err == nil {
-		if hostIP := net.ParseIP(hostname); hostIP != nil && validateHostIP(hostIP) == nil {
-			return hostIP
+	if rootIP == "" {
+		if hostname, err := os.Hostname(); err == nil {
+			if hostIP := net.ParseIP(hostname); hostIP != nil && validateHostIP(hostIP) == nil {
+				rootIP = hostIP.String()
+			}
 		}
 	}
 
 	// if that fails, check if we can find a primary IP address unambiguously
-	if allAddrs, err := net.InterfaceAddrs(); err == nil {
-		// adapted from: https://stackoverflow.com/a/31551220
-		for _, addr := range allAddrs {
-			if ipnet, ok := addr.(*net.IPNet); ok {
-				if validateHostIP(ipnet.IP) == nil {
-					return ipnet.IP
+	if rootIP == "" {
+		if allAddrs, err := net.InterfaceAddrs(); err == nil {
+			for _, addr := range allAddrs {
+				if ipnet, ok := addr.(*net.IPNet); ok {
+					if validateHostIP(ipnet.IP) == nil {
+						rootIP = ipnet.IP.String()
+						break
+					}
+				}
+			}
+		}
+	}
+
+	if rootIP == "" {
+		rootIP = net.IPv4(127, 0, 0, 1).String()
+		logrus.Warnf("unable to find a host IP, falling back to %v", rootIP)
+	}
+
+	// Search for an IPv6
+	ips := []string{rootIP}
```

Comment:
This should handle the case where rootIP is v6 but the interface has v4 addresses.

Reply:
I added the necessary code paths.